File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task6_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <vector>
#include <iostream>
#include <algorithm>

double findUniq(const std::vector<double> &v) {
    if (v[0] != v[1] && v[0] != v[2]) return v[0];
    return *std::find_if(v.begin(), v.end(), [&](double x) { return x != v[0]; });
}

int main() {
    std::cout << findUniq({1, 1, 1, 2, 1, 1}) << std::endl;  // Output: 2
    std::cout << findUniq({0, 0, 0.55, 0, 0}) << std::endl;  // Output: 0.55
    return 0;
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task6_CLAUDE_claude-3-5-sonnet-20240620.java

public class Kata {
    public static double findUniq(double[] arr) {
        if (arr[0] != arr[1] && arr[0] != arr[2]) return arr[0];
        if (arr[1] != arr[0] && arr[1] != arr[2]) return arr[1];
        
        for (int i = 2; i < arr.length; i++) {
            if (arr[i] != arr[0]) return arr[i];
        }
        
        return arr[0];
    }

    public static void main(String[] args) {
        System.out.println(findUniq(new double[]{ 1, 1, 1, 2, 1, 1 }));
        System.out.println(findUniq(new double[]{ 0, 0, 0.55, 0, 0 }));
    }
}
